src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/ConstantExpectedAnalyzer.UnmanagedHelper.cs

```diff
@@ -0,0 +1,350 @@
+            {
+                private readonly TransformHelper _helper;
+                public UnmanagedConstantExpectedParameter(IParameterSymbol parameter, T min, T max, TransformHelper helper) : base(parameter)
+                {
+                    Min = min;
+                    Max = max;
+                    _helper = helper;
+                }
+
+                public T Min { get; }
+                public T Max { get; }
+
+                public override bool ValidateParameterIsWithinRange(ConstantExpectedParameter subsetCandidate, IArgumentOperation argument, [NotNullWhen(false)] out Diagnostic? validationDiagnostics)
+                {
+                    if (Parameter.Type.SpecialType != subsetCandidate.Parameter.Type.SpecialType ||
+                        subsetCandidate is not UnmanagedConstantExpectedParameter subsetCandidateTParameter)
+                    {
+                        validationDiagnostics = Diagnostic.Create(CA1861.ConstantInvalidConstantRule, argument.Syntax.GetLocation(), Parameter.Type.ToDisplayString());
+                        return false;
+                    }
+
+                    if (!_helper.IsLessThan(subsetCandidateTParameter.Min, Min) && !_helper.IsLessThan(Max, subsetCandidateTParameter.Max))
+                    {
+                        //within range
+                        validationDiagnostics = null;
+                        return true;
+                    }
+                    validationDiagnostics = Diagnostic.Create(CA1861.ConstantOutOfBoundsRule, argument.Syntax.GetLocation(), Min.ToString(), Max.ToString());
+                    return false;
+                }
+
+                public override bool ValidateValue(IArgumentOperation argument, object? constant, [NotNullWhen(false)] out Diagnostic? validationDiagnostics)
+                {
+                    if (constant is not null && _helper.TryConvert(constant, out T value))
+                    {
+                        if (!_helper.IsLessThan(value, Min) && !_helper.IsLessThan(Max, value))
+                        {
+                            validationDiagnostics = null;
+                            return true;
+                        }
+                        validationDiagnostics = argument.CreateDiagnostic(CA1861.ConstantOutOfBoundsRule, Min.ToString(), Max.ToString());
+                        return false;
+                    }
+                    validationDiagnostics = argument.CreateDiagnostic(CA1861.ConstantInvalidConstantRule, Parameter.Type.ToDisplayString());
+                    return false;
+                }
+            }
+        }
+
+        private static bool TryConvertSignedInteger(object constant, out long integer)
+        {
+            try
+            {
+                if (constant is string or bool)
+                {
+                    integer = default;
+                    return false;
+                }
+                integer = Convert.ToInt64(constant);
```

Comment:
Is `constant` nullable? 
If yes, then that is fine.
If no, I think `long.Parse` would be more appropriate.

Reply:
Instead of the Try-Catch I also recommend to use TryParse.